lib/urbanopt/geojson.rb

```diff
@@ -57,7 +57,698 @@ def files_dir
+        all_pairs.each do |pair|
+          if point_is_shadowed(pair[:building_point], pair[:other_building_point], origin_lat_lon)
+            return true
+          end
+        end
+        return false
+      end
+
+
+      def point_is_shadowed(building_point, other_building_point, origin_lat_lon)
+      ##
+      # Returns Boolean indicating if specified building is shadowed
+      #
+      # Params:
+      # - building_point: nstance of OpenStudio::Point3d
+      # - other_building_point: other instance of OpenStudio::Point3d
+      # - origin_lat_lon: instance of OpenStudio::PointLatLon indicating origin lat & lon
+        vector = other_building_point - building_point
+        height = vector.z
+        distance = Math.sqrt(vector.x*vector.x + vector.y*vector.y)
+        if distance < 1
+          return true
+        end
+        hour_angle_rad = Math.atan2(-vector.x, -vector.y)
+        hour_angle = OpenStudio::radToDeg(hour_angle_rad)
+        lattitude_rad = OpenStudio::degToRad(origin_lat_lon.lat)
+        result = false
+        (-24..24).each do |declination|
+          declination_rad = OpenStudio::degToRad(declination)
+          zenith_angle_rad = Math.acos(Math.sin(lattitude_rad)*Math.sin(declination_rad) + Math.cos(lattitude_rad)*Math.cos(declination_rad)*Math.cos(hour_angle_rad))
+          zenith_angle = OpenStudio::radToDeg(zenith_angle_rad)
+          elevation_angle = 90-zenith_angle
+          apparent_angle_rad = Math.atan2(height, distance)
+          apparent_angle = OpenStudio::radToDeg(apparent_angle_rad)
+          if (elevation_angle > 0 && elevation_angle < apparent_angle)
+            result = true
+            break
+          end
+        end
+        return result
+      end
+
+
+      def floor_print_from_polygon(polygon, elevation, origin_lat_lon, zoning=false)
+      ##
+      # Returns Boolean indicating if specified building is shadowed
+      #
+      # Params:
+      # - polygon: array of coordinate pairs.
+      #   e.g. polygon = [
+              #   [1, 5],
+              #   [5, 5],
+              #   [5, 1],
+              # ]
+      # - elevation: number indicating elevation
+      # - origin_lat_lon: instance of OpenStudio::PointLatLon indicating origin lat & lon
+      # - zoning: zoning is true if you'd like to utilize aspects of function that are specific to zoning
+        floor_print = OpenStudio::Point3dVector.new
+        all_points = OpenStudio::Point3dVector.new
+        polygon.each do |p|
+          lon = p[0]
+          lat = p[1]
+          point_3d = origin_lat_lon.toLocalCartesian(OpenStudio::PointLatLon.new(lat, lon, 0))
+          point_3d = OpenStudio::Point3d.new(point_3d.x, point_3d.y, elevation)
+          curr_print = zoning ? OpenStudio::getCombinedPoint(point_3d, all_points, 1.0) : point_3d
+          floor_print << curr_print
+        end
+        if floor_print.size < 3
+          @runner.registerWarning("Cannot create floor print, fewer than 3 points")
```

Comment:
runner should be passed in as an argument to this function, it will be passed from the measure run method